### ISSUE TITLE
loops generators and out sockets are confused or even stop working

### ISSUE THREAD

**Issue by og76** [JacquesLucke/animation_nodes#260]
Precisely, (but not clear how..):
- if u have **generators with same type**: and **rename** them, the out sockets will provide switched data, generator A and B will go thru socket B and A etc
- u have to **plug and unplug the generator** inputs and again it goes confused
- by repating plug/unplug or rename, at a certain point it stops working (no output..)

+when creating another loop that has no connection with these, some issues get corrected (or on the contrary) and so on. On a big tree I had to give it up, and it was just math and indices mostly..

(I even got it workin depending on a boolean that was not connected to any generator yet, but that is a big tree and hard to follow..)

got this in several files:
https://drive.google.com/file/d/0BzzAoBwXO8dsbzRsVXJlZU1ndWM/view?usp=sharing
here, look at xyz, the debug goes in z, but the cube has -1 on x
switch the inputs of generators or rename generators to see more

https://drive.google.com/file/d/0BzzAoBwXO8dsdTRyb0tnX21tYkk/view?usp=sharing
here is the 3d template, duplicated the socket and added a transform to make a difference
if u switch the final list of vecs, you'll see that they are reversed
rename or plug unplug the generators for more fun
eventually you'll get into a point that does not work...

+it could be that not only same type goes crazy...
I'll try to repro more, but when reopen the files, sometimes they look different (in this regard)

**Comment by JacquesLucke** [comment]
Will still try out more but part of the problem was the sorting of the generator outputs.
Each generator output has a random number that will be used for sorting. In your case you duplicated the loop generator output 2 times -> all 3 nodes had the same sort index -> sort algorithm returns them in random order..

**Comment by og76** [comment]
ok, I'll try to further test
and yes, the renaming seamed to matter more as event than 

( could it be simpler to have a single output, like the group? )

**Comment by JacquesLucke** [comment]
Hm no, I don't think so I found it easier and cleaner to it like is now. Sorting of the outputs in the invoke subprogram node is really the only problem but I think I can solve that :)

Do the errors still appear or can we close this issue?
